devito/builtins/arithmetic.py

```diff
@@ -58,6 +58,42 @@ def norm(f, order=2):
     return f.dtype(v)
 
 
+@dv.switchconfig(log_level='ERROR')
+def sum(f, dims=None):
+    """
+    Compute the sum of the Function data over specified dimensions.
+    Defaults to sum over all dmensions
+
+    Parameters
+    ----------
+    f : Function
+        Input Function.
+    dims : Dimension or tuple
+        Dimensions to sum over
+    """
+    if dims is None or dims == f.dimensions:
+        return sumall(f)
+
+    dims = dv.tools.as_tuple(dims)
+    new_dims = tuple(d for d in f.dimensions if d not in dims)
+    shape = tuple(f._size_domain[d] for d in new_dims)
+    if f.is_TimeFunction and f.time_dim not in dims:
+        out = dv.TimeFunction(name="%ssum" % f.name, grid=f.grid,
```

Comment:
I've recently implemented `_rebuild` see [here](https://github.com/devitocodes/devito/blob/master/devito/passes/iet/asynchrony.py#L130) for example

so you should be able to just do: 
```
f._rebuild(name=..., shape=shape, dimensions=new_dims)
```

Reply:
In some cases yeah but not all. There is still corner cases (like sparse) I need to add and because the class can change depending on the dimension summed rebuild won't work. (E.g summing a SparseTime over the rec dim makes a TimeFunc not a rebuild)

Reply:
sure, but why not using it where possible?

Reply:
Was erroring, some kwargs missing haven't had time to dig into it